shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/dml/InsertStatement.java

```diff
@@ -95,107 +76,63 @@
         return Optional.ofNullable(insertSelect);
     }
     
-    /**
-     * Get on duplicate key columns segment.
-     *
-     * @return on duplicate key columns segment
-     */
-    public Optional<OnDuplicateKeyColumnsSegment> getOnDuplicateKeyColumns() {
-        return Optional.ofNullable(onDuplicateKeyColumns);
-    }
-    
-    /**
-     * Get with segment.
-     * 
-     * @return with segment.
-     */
-    public Optional<WithSegment> getWithSegment() {
-        return Optional.ofNullable(withSegment);
-    }
-    
     /**
      * Judge is use default columns or not.
      * 
      * @return is use default columns or not
      */
```

Comment:
Hi, I am glad to see this PR raized since it is the`real` refactoring work touching some of the database dialect issues.
Separately reviewing this PR, I suppose it seems great except for a renaming issue. However, we can not avoid handling database dialects later. Therefore I want to give two points after my deliberation. Could you give them some consideration?

1. A interface named `SQLstatementUtil<T entends SQLStatement>` (Like ·SQLStatementContext<T extends SQLStatement>·) is needed to deal with the judgment or handling for different dialect SQLStatments in the process of rewriting, routing, etc. 

2. We should keep all the `SQLStatment` clear and simple enough, which means `SQLStatement` just provides the basic public functions. 

**What's the difference between `SQLStatement` and `SQLStatmentContext`. ?**

My answer is that `SQLStatment` is the parsed `OBJECT` result for SQL input. Hence, `SQLStatment` is supposed to publish elementary functions as possible. On the other hand, `SQLStatmentContext` will carry all the logic handlings for ShardingSphere using `SQLStatment`.

Based on that, some functions like `getValueCountForPerGroup` in `InsertStatment` maybe belongs to `SQLStatmentContext` instead of `SQLStatement`. We can consider improving it later. Plus, not only this one, other `SQLStatment's are also in our consideration. Actually speaking, this is skilled work. Thus  I don't feel comfortable letting others take charge of this task. But if you had an interest in it,  then welcome. :-)

Reply:
@tristaZero Thank you very much, I think your suggestions are great. 👏 `InsertStatment` really should not be overly complicated, and methods such as `getValueCountForPerGroup` are also usually used in `SQLStatmentContext`, so these methods should be migrated to `SQLStatmentContext`. I will try to refactor `SQLStatement` code to make it more elegant. 😄